R/orderlyweb_run.R

```diff
@@ -33,25 +33,41 @@ report_run_wait <- function(path, name, key, client, timeout, poll, open,
 }
 
 
-report_run_query <- function(ref, update, timeout, instance) {
-  query <- list()
-  if (!is.null(ref)) {
-    query$ref <- ref
-  }
-  if (!update) {
-    query$update <- "false"
-  }
+report_run_query <- function(timeout) {
   if (!is.null(timeout)) {
     assert_scalar_integer(timeout)
-    query$timeout <- as.character(timeout)
+    query <- list(
+      timeout = as.character(timeout)
+    )
+  } else {
+    query <- NULL
+  }
+  query
+}
+
+report_run_body <- function(parameters, ref, instance) {
+  body <- list()
+  if (!is.null(ref)) {
+    assert_scalar_character(ref)
+    body$gitCommit <- ref
```

Comment:
The reason the tests were failing before is because orderly-web has changed the expected format of report run request. See https://github.com/vimc/orderly-web/blob/master/docs/spec/spec.md#post-reportsnamerun
This is support the report runner interface.
The camel case gitCommit looks a bit strange as elsewhere the API is using snake case - we had this same tension with naomi too where they wanted to use camel case and we wanted snake case.

Reply:
You can make this look slightly less weird by doing

```
body[["gitCommit"]] <- scalar(ref)
```

which makes it look more like building up data which is what you're doing. But it's equivalent.

Do we need to mark these as scalars or are we unboxing automatically?

Reply:
Thanks using the double `[` looks nicer. We're using the automatic unboxing in `httr`